poc/flp_generic.py

```diff
@@ -640,6 +641,137 @@ def test_vec_set_type_param(self, test_vec):
+    JOINT_RAND_LEN = 2
+    OUTPUT_LEN = None  # Set by constructor
+
+    def __init__(self, length, max_count, chunk_length):
+        """
+        Instantiate an instance of the `MultiHotHistogram` circuit with the
+        given length, max_count, and chunk_length.
+        """
+        if length <= 0:
+            raise ValueError('invalid length')
+        if max_count <= 0 or max_count > length:
+            raise ValueError('invalid max_count')
+        if chunk_length <= 0:
+            raise ValueError('invalid chunk_length')
+
+        # Compute the number of bits to represent `max_count`.
+        self.bits_for_count = int(math.log2(max_count)) + 1
+        self.offset = self.Field((1 << self.bits_for_count) - 1 - max_count)
+        # Sanity check `offset + length` doesn't overflow field size,
+        # because in validity circuit, we will compute `offset + \sum_i meas_i`.
+        if self.Field.MODULUS - self.offset.as_unsigned() <= length:
+            raise ValueError('length and max_count are too large '
+                             'for the current field size')
+
+        self.length = length
+        self.max_count = max_count
+        self.chunk_length = chunk_length
+        self.GADGETS = [ParallelSum(Mul(), chunk_length)]
+        # The number of bit entries are `length + bits_for_count`,
+        # so the number of gadget calls is equal to
+        # `ceil((length + bits_for_count) / chunk_length)`.
+        self.GADGET_CALLS = [
+            (length + self.bits_for_count + chunk_length - 1) // chunk_length
+        ]
+        self.MEAS_LEN = self.length + self.bits_for_count
+        self.OUTPUT_LEN = self.length
+
+    def eval(self, meas, joint_rand, num_shares):
+        self.check_valid_eval(meas, joint_rand)
+
+        # Check that each bucket is one or zero.
+        range_check = self.Field(0)
+        r = joint_rand[0]
+        r_power = r
+        shares_inv = self.Field(num_shares).inv()
+        for i in range(self.GADGET_CALLS[0]):
+            inputs = [None] * (2 * self.chunk_length)
+            for j in range(self.chunk_length):
+                index = i * self.chunk_length + j
+                if index < len(meas):
+                    meas_elem = meas[index]
+                else:
+                    meas_elem = self.Field(0)
+
+                inputs[j * 2] = r_power * meas_elem
+                inputs[j * 2 + 1] = meas_elem - shares_inv
+
+                r_power *= r
+
+            range_check += self.GADGETS[0].eval(self.Field, inputs)
+
+        # Check that `offset` plus the sum of the buckets is equal to the
+        # value claimed by the Client.
+        count_check = self.offset * shares_inv
+        for i in range(self.length):
+            count_check += meas[i]
+        count_check -= self.Field.decode_from_bit_vector(meas[self.length:])
+
+        out = joint_rand[1] * range_check + \
+            joint_rand[1] ** 2 * count_check
+        return out
```

Comment:
```suggestion
        return joint_rand[1]    * range_check + \
               joint_rand[1]**2 * count_check
```

Reply:
We have `autopep8` in our linter CI job, and it won't like the irregular number of spaces around operators. We could add some extra parentheses if readability is of concern.

Reply:
:(((

No no, let's do what the robot tells us to do. The main thing is to be consistent I think.